h/api_client/subscribers.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+import pyramid.events
+
+import h.api_client.api_client as api_client
+
+
+def _get_api_url(request):
+    """Return the configured base URL for this app's Hypothesis API instance.
+
+    Returns the value of the h.api_url setting from the config file,
+    or if that doesn't exist the "/api" route from the WSGI app (which assumes
+    that we're running the app with the "api" feature enabled).
+
+    Always returns the URL _without_ a trailing /, despite what the user might
+    have entered in the config file.
+
+    """
+    return request.registry.settings.get(
+        "h.api_url", request.resource_url(request.root, "api")).rstrip("/")
```

Comment:
I'm not sure this is necessary if there's already a way of retrieving a configured API client from the request. The API root URL in the settings should be the concern of only one component in the system -- the API client. I don't think we want to go hanging extra properties of the request just in case we find another use for them.